src/analysis/processing/qgsalgorithmcheckgeometryangle.cpp

```diff
@@ -76,27 +77,36 @@ void QgsGeometryCheckAngleAlgorithm::initAlgorithm( const QVariantMap &configura
       QList<int>() << static_cast<int>( Qgis::ProcessingSourceType::VectorPolygon ) << static_cast<int>( Qgis::ProcessingSourceType::VectorLine )
     )
   );
-  addParameter( new QgsProcessingParameterField( QStringLiteral( "UNIQUE_ID" ), QObject::tr( "Unique feature identifier" ), QString(), QStringLiteral( "INPUT" ) ) );
-  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "MIN_ANGLE" ), QObject::tr( "min angle" ), Qgis::ProcessingNumberParameterType::Double, 0, false, 0.0, 180.0 ) );
+  addParameter( new QgsProcessingParameterField(
+    QStringLiteral( "UNIQUE_ID" ), QObject::tr( "Unique feature identifier" ), QString(), QStringLiteral( "INPUT" )
+  ) );
+  addParameter( new QgsProcessingParameterNumber(
+    QStringLiteral( "MIN_ANGLE" ), QObject::tr( "Minimum angle" ), Qgis::ProcessingNumberParameterType::Double, 0, false, 0.0, 180.0
+  ) );
 
   // outputs
-  addParameter( new QgsProcessingParameterFeatureSink( QStringLiteral( "ERRORS" ), QObject::tr( "Error layer" ), Qgis::ProcessingSourceType::VectorPoint ) );
-  addParameter( new QgsProcessingParameterFeatureSink( QStringLiteral( "OUTPUT" ), QObject::tr( "Output layer" ), Qgis::ProcessingSourceType::VectorAnyGeometry ) );
-
-  auto tolerance = std::make_unique<QgsProcessingParameterNumber>( QStringLiteral( "TOLERANCE" ), QObject::tr( "Tolerance" ), Qgis::ProcessingNumberParameterType::Integer, 8, false, 1, 13 );
+  addParameter( new QgsProcessingParameterFeatureSink(
+    QStringLiteral( "ERRORS" ), QObject::tr( "Error layer" ), Qgis::ProcessingSourceType::VectorPoint
+  ) );
+  addParameter( new QgsProcessingParameterFeatureSink(
+    QStringLiteral( "OUTPUT" ), QObject::tr( "Output layer" ), Qgis::ProcessingSourceType::VectorAnyGeometry
+  ) );
+
+  std::unique_ptr<QgsProcessingParameterNumber> tolerance = std::make_unique<QgsProcessingParameterNumber>(
+    QStringLiteral( "TOLERANCE" ), QObject::tr( "Tolerance" ), Qgis::ProcessingNumberParameterType::Integer, 8, false, 1, 13
```

Comment:
As mentioned in https://github.com/qgis/QGIS-Documentation/pull/9773, afaiu I find this tolerance parameter (_8 meaning 10^-8_) non-obvious, and its min/max values counter-intuitive. A parameter help is at least required for end users, but imho, we should move to something simpler as a plain double value with unit selector, as commonly used for this kind of parameter.

Reply:
Yes but it's to keep consistency with Geometry Checker plugin:


https://github.com/user-attachments/assets/19d7327c-0c7c-4d07-bd29-35a51ebd8f1c

Reply:
btw, Geometry Checker accepts only an integer https://github.com/qgis/QGIS/blob/master/src/analysis/vector/geometry_checker/qgsgeometrycheckcontext.cpp#L20

Reply:
> Yes but it's to keep consistency with Geometry Checker plugin:

Probably, but 1E-8 as shown in the plugin looks much more meaningful than 8. I don't know how strong is/should be the link between the two tools and I'm not speaking in terms of code, rather as a Processing end user who would expect some obviousness in the options and harmonization (and that we take advantage from improved parameter types in Processing)...

Reply:
Yes I agree that from an end user perspective, it could be confusing. However, we are tied with how the geometry checker is written right now.

We can't use the distance widget parameter because it would be nonsense and difficult to restrict the user to specific logarithmic values like "10, 1, 0.1, 0.001, 0.0001, etc.".

Our options are 
1. to create another parameter type, think of its UI, and that handles the differences we have between geographic and metric distances
2. to change how the geometry checker tolerance is given

I prefer to go with option 2. but in any case, it will be a work that would be done in a dedicated PR.

Right now, we could improve the documentation, or add a description of the tolerance parameter in every processing description.

Reply:
> Right now, we could improve the documentation, or add a description of the tolerance parameter in every processing description.

I would have said 
> Right now, we could improve the documentation, ~~or~~**AND** add a description of the tolerance parameter in every processing description.

They should have a start of answer in the GUI imho.

Reply:
Added a short text in every processing short help.
> The "Tolerance" advanced parameter defines the numerical precision of geometric operations, given as an integer n, meaning that any difference smaller than 10⁻ⁿ (in map units) is considered zero.

Reply:
Given that this option is in advanced panel, I doubt people will modify it and don't think it deserves a mention in the algorithm main help. It should instead be as a tooltip over the parameter only, i.e.:
```
  tolerance->setHelp( QObject::tr( "The \"Tolerance\" defines the numerical precision of geometric operations, "
                                   "given as an integer n, meaning that any difference smaller than 10⁻ⁿ (in map units) is considered zero." );
```